server/channels/store/localcachelayer/user_layer.go

```diff
@@ -216,6 +216,11 @@ func (s *LocalCacheUserStore) GetProfileByIds(ctx context.Context, userIds []str
 	return users, nil
 }
 
+func (s *LocalCacheUserStore) UpdateFailedPasswordAttempts(userID string, attempts int) error {
+	s.InvalidateProfileCacheForUser(userID)
```

Comment:
If we invalidate here, then we don't need to invalidate in the app layer. Let's choose only one thing.

Reply:
I think ideally, every write operation should automatically clear the appropriate cache(if any) for that write operation. 
This will free the client from the burden of _identifying_ which cache to clear and also free them from _making sure_ to clear the cache _every time_ they call a write operation.

And you are right, since there is no other write operation in the app layer. We can just remove _both_ calls to invalidateCache.